pkg/ruler/compat.go

```diff
@@ -63,7 +63,7 @@ type PusherAppendable struct {
 }
 
 // Appender returns a storage.Appender
-func (t *PusherAppendable) Appender() storage.Appender {
+func (t *PusherAppendable) Appender(_ context.Context) storage.Appender {
```

Comment:
I think we should propagate context from here to `Pusher.Push()`. It will also help to remove `userID` parameter from `pusherAppender` struct. We can encode the userID into context, in ruler.go here https://github.com/cortexproject/cortex/blob/a01c1012643a8afbc259950db659e4a9e587259c/pkg/ruler/ruler.go#L543 
.. and extract userID from context in this file as part of `DefaultTenantManagerFactory`

Reply:
> It will also help to remove userID parameter from pusherAppender struct. We can encode the userID into context, in ruler.go here

In other areas of the code, we prefer to pass userID explicitly, and not hide it into Context. But Cortex codebase is definitely not consistent about it :(

Reply:
Oh, thanks @pstibrany! Would you prefer we separate it out? Its definitely a by-product in this PR, its more important we pass context along :)

Reply:
> Would you prefer we separate it out?

Yes, I think it would be bit cleaner approach. Thanks! (And sorry for not reacting earlier)

Reply:
No worries @pstibrany :)

Reply:
Yes, let's be explicit here and pass `userID` as parameter.